pandas/core/categorical.py

```diff
@@ -1085,10 +1085,14 @@ def searchsorted(self, v, side='left', sorter=None):
                              "ordered one")
 
         from pandas.core.series import Series
-        values_as_codes = self.categories.values.searchsorted(
-            Series(v).values, side=side)
+        values_as_codes = _get_codes_for_values(Series(v).values,
+                                                self.categories)
 
-        return self.codes.searchsorted(values_as_codes, sorter=sorter)
+        if -1 in values_as_codes:
```

Comment:
see here: https://github.com/pandas-dev/pandas/blob/master/pandas/computation/pytables.py#L203. The idea IS to use searchsorted. Then check the 0's (only). If they are not actual categories, then you would raise.